filebeat/prospector/prospector_log_test.go

```diff
@@ -62,3 +64,96 @@ func (t TestFileInfo) Mode() os.FileMode  { return 0 }
 func (t TestFileInfo) ModTime() time.Time { return t.time }
 func (t TestFileInfo) IsDir() bool        { return false }
 func (t TestFileInfo) Sys() interface{}   { return nil }
+
+func TestGlob(t *testing.T) {
+	var tests = []struct {
+		pattern         string
+		expectedMatches []string
+	}{
+		{
+			"*",
+			[]string{
+				"foo",
+			},
+		},
+		{
+			"foo/*",
+			[]string{
+				"foo/bar",
+			},
+		},
+		{
+			"*/*",
+			[]string{
+				"foo/bar",
+			},
+		},
+		{
+			"**",
+			[]string{
+				"foo",
+				"foo/bar",
+			},
+		},
+		{
+			"foo**",
+			[]string{
+				"foo",
+			},
+		},
+		{
+			"foo/**",
+			[]string{
+				"foo/bar",
+				"foo/bar/baz",
+			},
+		},
+		{
+			"foo/**/baz",
+			[]string{
+				"foo/bar/baz",
+			},
+		},
+		{
+			"foo/**/bazz",
+			[]string{},
+		},
```

Comment:
some more tests to add: 
- double `/` like `foo//bar`
- `foo\bar`
- `foo\**\bar`
- `c:\foo\**\bar`